### PR TITLE
Malus feature

### LINKED ISSUE #9: Malus object

### PR DESCRIPTION
**Pull request by ValentinaPieri**
Closes #9

### CODE REVIEW

File: src/main/state_changers/BlackStain.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import javax.imageio.ImageIO;
+import javax.swing.JPanel;
+import javax.swing.Timer;
+
+public class BlackStain extends Malus implements ActionListener{
+	
+	//Dirty Screen
+	public static final Dimension SIZE = Toolkit.getDefaultToolkit().getScreenSize();
+	private Image blackstains;
+	StainPanel stain = new StainPanel(blackstains);
```
Comment (expl0r3rgu1) [inline src/main/state_changers/BlackStain.java:22]: Here you are giving a null Image to new StainPanel()
You should first initialize blackstains and then stain. Both in the constructor.

File: src/main/state_changers/BlackStain.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	public Object changeState() {
+		timer.start();
+		play.add(stain);
+		
+		return null;
+	}
+	
+	public void animate(Graphics2D canvas) {
+		canvas.drawImage(getSkin().getImage(), position.getX(), position.getY(), SIZE.getWidth() * 3 / 100, SIZE.getWidth() * 3 / 100, null);
```
Comment (expl0r3rgu1) [inline src/main/state_changers/BlackStain.java:37]: position never gets updated and should also be called through the getter getPosition()

File: src/main/state_changers/BlindBlock.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import javax.swing.JPanel;
+import javax.swing.Timer;
+import main.utilities.Skin;
+
+public class BlindBlock extends Malus implements ActionListener{
+	
+	//Black Block
+	public static final Dimension SIZE = Toolkit.getDefaultToolkit().getScreenSize();
+	private Image blindBlock;
+	StainPanel stain = new StainPanel(blindBlock);
```
Comment (expl0r3rgu1) [inline src/main/state_changers/BlindBlock.java:22]: Same issues as with BlackStain here

File: src/main/state_changers/CoinsReducer.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	public CoinsReducer(Position position, Skin skin) {
+		super(position, skin);
+	}
+
+	public Integer changeState() {
+		return malus.get((int) (Math.random()*(malus.size())));
+	}
+	
+	public void animate(Graphics2D canvas) {
+		canvas.drawImage(getSkin().getImage(), position.getX(), position.getY(), SIZE.getWidth() * 3 / 100, SIZE.getWidth() * 3 / 100, null);
```
Comment (expl0r3rgu1) [inline src/main/state_changers/CoinsReducer.java:26]: position not updated and not called through the getter getPosition()

**Review by expl0r3rgu1 — CHANGES_REQUESTED** [review]

File: src/main/state_changers/CoinsReducer.java (hunk 4 above)
Comment (ValentinaPieri) [reply src/main/state_changers/CoinsReducer.java]: Done

File: src/main/state_changers/BlindBlock.java (hunk 3 above)
Comment (ValentinaPieri) [reply src/main/state_changers/BlindBlock.java]: Done

File: src/main/state_changers/BlackStain.java (hunk 2 above)
Comment (ValentinaPieri) [reply src/main/state_changers/BlackStain.java]: Done

File: src/main/state_changers/BlackStain.java (hunk 1 above)
Comment (ValentinaPieri) [reply src/main/state_changers/BlackStain.java]: Done

**Review by criicchio — APPROVED** [review]

File: src/main/state_changers/StainPanel.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+		this.setOpaque(false);
+		
+		this.setVisible(true);
+	}
+	
+	@Override
+	protected void paintComponent(Graphics g) {
+		super.paintComponent(g);
+		Graphics2D canvas = (Graphics2D) g;
+		canvas.drawImage(scale(image), 0, 0, (int) SIZE.getWidth(), (int) SIZE.getHeight(), null);
```
Comment (expl0r3rgu1) [inline src/main/state_changers/StainPanel.java:38]: Remove redundant scale method

**Review by expl0r3rgu1 — CHANGES_REQUESTED** [review]

**Review by benedettapacilli — APPROVED** [review]

File: src/main/state_changers/StainPanel.java (hunk 5 above)
Comment (ValentinaPieri) [reply src/main/state_changers/StainPanel.java]: Done

**Review by expl0r3rgu1 — APPROVED** [review]

**Pull request merged**